### ISSUE TITLE
How does the final sha256 prevent pass-the-hash?

### ISSUE THREAD

**Issue by rfjakob** [dxa4481/clientHashing#1]
Maybe I am missing something, but what you call "user_hash" in the flow chart can be passed around just fine?

**Comment by dxa4481** [comment]
The user_hash is not stored anywhere, at any time. In the event of a database breach, you do not obtain values that help you to log into anything. 

Contrasted with Microsoft Windows NTLM authentication, when you compromise one machine, you can steal the hash off that system, and use it to then authenticate to other machines.

**Comment by rfjakob** [comment]
Yes, but you can capture it off the network. And I think this contradicts
what you state in the readme,

> Cleartext credentials (over TLS) are also never transmitted to the server;

**Comment by dxa4481** [comment]
You can capture the hash if you have a passive man in the middle, and replay it. That's correct. That doesn't contradict what's in the readme, you don't get the cleartext credential out of it. In a normal workflow, this type of man in the middle would allow you to get the cleartext credential, which is not only replayable to the application, but due to password re-use, replayable to other applications as well.
